### PR TITLE
Feature/multi user signing frontend

### PR DESCRIPTION
**Pull request by dhelmert**
### Reason for this PR
- The operation management is needed for Multi-User-Signing, to handle access rights, operation logic and caching

### Changes in this PR
- Add operation-management.yaml with endpoints for fetching operations with various filters

### CODE REVIEW

File: operation-management.yaml [hunk 1]
```diff
@@ -0,0 +1,301 @@
+openapi: "3.0.0"
+info:
+  description: "This is the Operation API for UC4."
+  version: "0.12.1"
```
Comment (bastihav) [inline operation-management.yaml]:
```suggestion
  version: "0.15.1"
```

File: operation-management.yaml [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+  title: "UC4"
+
+servers:
+  - url: https://uc4.cs.upb.de/api/operation-management
```
Comment (bastihav) [inline operation-management.yaml]:
```suggestion
  - url: https://uc4.cs.upb.de/api/production/operation-management
```

File: operation-management.yaml [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+  externalDocs:
+    description: "Find out more"
+    url: "https://github.com/upb-uc4"
+- name: "Operation Management"
+  description: "Everything about the Operations"
+  externalDocs:
+    description: "Find out more"
+    url: "https://github.com/upb-uc4"
+
+paths:
```
Comment (bastihav) [inline operation-management.yaml:24]: hl version missing

File: operation-management.yaml [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+      parameters:
+      - name: "selfInitiated"
+        in: "query"
+        description: "Flag that determines, if only operations initiated by yourself are fetched"
+        required: false
+        schema:
+          type: 'boolean'
+      - name: "selfActionRequired"
+        in: "query"
+        description: "Flag that determines, if only operations that you can approve/deny are fetched"
```
Comment (bastihav) [inline operation-management.yaml]:
```suggestion
        description: "Flag that determines, if only operations that you can approve/reject are fetched"
```

File: operation-management.yaml [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+        required: false
+        schema:
+          type: 'boolean'
+      - name: "states"
+        in: "query"
+        description: "List of states to filter by (PENDING, FINISHED, or REJECTED)"
```
Comment (bastihav) [inline operation-management.yaml]:
```suggestion
        description: "Comma-seperated list of operation states to filter by (PENDING, FINISHED, or REJECTED)"
```

File: operation-management.yaml [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/GenericError'
+  /operations/{operationId}:
+    get:
+      tags:
+      - "Operation Management"
+      summary: "Find Operation by OperationId"
+      description: "Find course by OperationId"
```
Comment (bastihav) [inline operation-management.yaml]:
```suggestion
      description: "Find operation by OperationId"
```

**Review by bastihav — APPROVED** [review]

File: operation-management.yaml (hunk 3 above)
Comment (dhelmert) [reply operation-management.yaml:24]: Has been added as of now

**Review by bastihav — APPROVED** [review]

**Review by bastihav — CHANGES_REQUESTED** [review]
examples for matriculation payload are wrong due to approve/reject restructuring

**Comment by dhelmert** [comment]
> 
> 
> examples for matriculation payload are wrong due to approve/reject restructuring

This is a minimal working api, the new changes, for which these examples are wreong will be coming in another branch

**Review by bastihav — APPROVED** [review]

**Pull request merged**